### PR TITLE
SiteGroups Personabar Extension

### PR DESCRIPTION
**Pull request by donker**
## Summary
DNN uses SiteGroups "under the hood" to allow you to group portals together and use a single one for user authentication. This part of the API had no front end UI, however. This is a personabar extension created originally by @SCullman and kindly donated. I integrated it and resolved a few minor bugs.

### CODE REVIEW

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/SiteGroups/PortalGroupAdapter.cs [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                    var p = new PortalController().GetPortal(portal.PortalId);
+                    PortalGroupController.AddPortalToGroup(p, @group, callback);
+                }
+            }
+            return @group.PortalGroupId;
+        }
+
+        public void Delete(int portalGroupId)
+        {
+            Trace.TraceInformation("Delete PortalGroup");
```
Comment (bdukes) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/SiteGroups/PortalGroupAdapter.cs:130]: Should this be using DNN's logging framework?

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/SiteGroups/PortalGroupAdapter.cs [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        IEnumerable<DotNetNuke.Entities.Portals.PortalInfo> PortalsOfGroup(int groupId, int masterPortalId)
+        {
+            return PortalGroupController
+                .GetPortalsByGroup(groupId)
+                .Where(x => x.PortalID != masterPortalId);
+        }
+
+        public int Save(PortalGroupInfo portalGroup)
+        {
+            Trace.TraceInformation("Save PortalGroup");
```
Comment (bdukes) [inline Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/SiteGroups/PortalGroupAdapter.cs:68]: As below, this seems like it should be using DNN's log

File: Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/SiteGroups/PortalGroupAdapter.cs (hunk 1 above)
Comment (donker) [reply Dnn.AdminExperience/Dnn.PersonaBar.Extensions/Components/SiteGroups/PortalGroupAdapter.cs:130]: Good catch. But that should happen in the controller, not in the front end. I'll work on that.

**Comment by bdukes** [comment]
I've rebased on `develop`.  Once the build is done, this can be merged.

**Comment by SCullman** [comment]
Thank you all for taking action!

**Comment by valadas** [comment]
Thanks @donker for taking care of this item, awesomeness! One less thing in my very long todo list :)

And thanks @SCullman for donating this.

**Pull request merged**